### DISCUSSION TITLE
Cant add row to MariaDB

### DISCUSSION THREAD

**Discussion by mrjoli021** [sqlalchemy/sqlalchemy#7042] [Usage Questions]
I am using sqlalchemy with MariaDB.  I am trying to follow the guide provided 
https://mariadb.com/resources/blog/using-sqlalchemy-with-mariadb-connector-python-part-1/
I am modifying my code to make it a restful api using flask.  When I try to add a new employee I get the following when I send firstname, lastname, and active via postman:

Not sure what I am missing.

add_employee
    newEmployee = Employee(firstname, lastname, active)
TypeError: __init__() takes 1 positional argument but 4 were given

`from flask import Flask, request, jsonify
import sqlalchemy
from sqlalchemy.ext.declarative import declarative_base
from flask_sqlalchemy import SQLAlchemy
import os
import json

#Init app
app = Flask(__name__)

#Gets password info from json file
PASSWORD_FILE = open('password.json')
DATA = json.load(PASSWORD_FILE)

#Contant Data
DB = DATA['DB']
DB_HOST = DATA['Host']
DB_USER = DATA['User']
DB_PASS = DATA['Pass']
DB_PORT = DATA['Port']

# Define the MariaDB engine using MariaDB Connector/Python
engine = sqlalchemy.create_engine(f"mariadb+pymysql://{DB_USER}:{DB_PASS}@{DB_HOST}:{DB_PORT}/{DB}")

Base = declarative_base()

class Employee(Base):

   __tablename__ = 'employees'
   id = sqlalchemy.Column(sqlalchemy.Integer, primary_key=True)
   firstname = sqlalchemy.Column(sqlalchemy.String(length=100))
   lastname = sqlalchemy.Column(sqlalchemy.String(length=100))
   active = sqlalchemy.Column(sqlalchemy.Boolean, default=True)

#creates db model
Base.metadata.create_all(engine)

#initialize Session
Session = sqlalchemy.orm.sessionmaker()
Session.configure(bind=engine)
session = Session()

#create Employee
@app.route('/', methods=['POST'])
def add_employee():
    firstname = request.json['firstname']
    lastname = request.json['lastname']
    active = request.json['active']

    newEmployee = Employee(firstname, lastname, active)
    session.add(newEmployee)
    session.commit()

    return employee_schema.jsonify(new_employee)

#Run Server
if __name__ == '__main__':
    app.run(debug=True)
`

**Comment by CaselIT** [comment]
Hi,

The issue seems to be in the Employee definition, but since it's not in your snippet that just a guess
